spring-cloud-kubernetes-fabric8-discovery/src/main/java/org/springframework/cloud/kubernetes/fabric8/discovery/KubernetesDiscoveryClient.java

```diff
@@ -162,13 +162,13 @@ private List<ServiceInstance> getNamespaceServiceInstances(EndpointSubsetNS es,
 
 	@Override
 	public List<String> getServices() {
```

Comment:
if we have for example enabled `all-namespaces=true` and there is such instances:

- `my-service` in namespace `a`
- `my-service` in namespace `b`

a call to `getServices` will currently return `[my-service, my-service]`, which imo, does not make sense to the callers. wdyt?